In this report, we will first present our package which contains the process_math part and the process_vis one, which appeals to modular arithmetic, the notion of graph and also the graphical aspects. 

At the start, we will create the associated undirected graph in the console for the parameters: table_numer is equal to 2, and modulo_number = 400

In [1]:
%pip install multiplication_table
import multiplication_table as mt
import time
from tkinter import Tk, Canvas

table_number1=2
modulo_number1 = 400

graph_object1 = mt.Graph(table_number1, modulo_number1)
graph_object1.sparse_matrix()
graph_object1.print_graph()

You should consider upgrading via the 'C:\Users\Nouhaila\anaconda3\python.exe -m pip install --upgrade pip' command.

sparse_matrix'time0.0009980201721191406
sparse_matrix'time0.0
1 <---> 2.0
2 <---> 4.0
3 <---> 6.0
4 <---> 8.0
5 <---> 10.0
6 <---> 12.0
7 <---> 14.0
8 <---> 16.0
9 <---> 18.0
10 <---> 20.0
11 <---> 22.0
12 <---> 24.0
13 <---> 26.0
14 <---> 28.0
15 <---> 30.0
16 <---> 32.0
17 <---> 34.0
18 <---> 36.0
19 <---> 38.0
20 <---> 40.0
21 <---> 42.0
22 <---> 44.0
23 <---> 46.0
24 <---> 48.0
25 <---> 50.0
26 <---> 52.0
27 <---> 54.0
28 <---> 56.0
29 <---> 58.0
30 <---> 60.0
31 <---> 62.0
32 <---> 64.0
33 <---> 66.0
34 <---> 68.0
35 <---> 70.0
36 <---> 72.0
37 <---> 74.0
38 <---> 76.0
39 <---> 78.0
40 <---> 80.0
41 <---> 82.0
42 <---> 84.0
43 <---> 86.0
44 <---> 88.0
45 <---> 90.0
46 <---> 92.0
47 <---> 94.0
48 <---> 96.0
49 <---> 98.0
50 <---> 100.0
51 <---> 102.0
52 <---> 104.0
53 <---> 106.0
54 <---> 108.0
55 <---> 110.0
56 <---> 112.0
57 <---> 114.0
58 <---> 116.0
59 <---> 118

And then, for the graphical interface, we made the link between the different aspects of the visualization, as well as design and movement. 

In [2]:
start = time.time()
root = Tk()
canvas = Canvas(root, width=750, height=750, bg='white')
canvas.pack(side="left", fill="both", expand=True)
mt.circle(canvas=canvas, center=(360, 360), radius=300,
          state_circle=True, background_circle="", outline_circle="black")
mt.dot(canvas=canvas, graph=mt.Graph(2, 10), radius=300, center=360,
       color_graph=['black', 'purple', "blue", "red", "cyan"],
       color_name="red")
mt.all_edges(canvas=canvas, graph=mt.Graph(2, 10), radius=300, center=360,
             color_graph=['black', 'purple', "blue", "red", "cyan"],
             edges_width=2)
root.mainloop()
end = time.time()
print("Fixe representation 'time : " + str(end-start))

sparse_matrix'time0.0
sparse_matrix'time0.0
Fixe representation 'time : 2.632767915725708


Looking at the execution time, we notice that it is equal to 2.632767915725708

In [3]:
mt.Interface_gestion(speed=200, state_button=False, background='white',
                     state_circle=False,
                     color_graph=['purple', "blue", "orange", "brown"],
                     background_circle="", outline_circle='black',
                     color_name="red", edges_width=1)

sparse_matrix'time0.0
graph_vis'time : 0.007979154586791992
graph_vis'time : 0.025931835174560547
